app/cdap/components/Connections/Browser/GenericBrowser/index.tsx

```diff
@@ -116,7 +119,11 @@ export function GenericBrowser({ selectedConnection }) {
     const { name: entityName, canBrowse } = entity;
     if (!canBrowse) {
       setLoading(true);
-      return onCreateWorkspace(entity);
+      getEntitySpec(entity);
+
+      return typeof onEntitySelect === 'function'
+        ? getEntitySpec(entity)
+        : onCreateWorkspace(entity);
```

Comment:
onCreateWorkspace isn't a function name I'd expect to return a value. What kind of value does it return? If it's a promise with no return value, I think it would be clearer to do:
```tsx
const onExplore = async (entity) => {
  ...;
  if (typeof onEntitySelect === 'function') {
    await getEntitySpec(entity);
  } else {
    await onCreateWorkspace(entity);
  }
}
```